mypy/semanal.py

```diff
@@ -1848,7 +1850,19 @@ def analyze_lvalue(self, lval: Lvalue, nested: bool = False,
                 self.type.names[lval.name] = SymbolTableNode(MDEF, v)
             elif explicit_type:
                 # Don't re-bind types
-                self.name_already_defined(lval.name, lval)
+                global_def = self.globals.get(lval.name)
+                if self.locals:
+                    locals_last = self.locals[-1]
```

Comment:
@gvanrossum probably will be happy to see another case in mypy for PEP 572 :-)

@sid-kap You can ignore this comment.